### ISSUE TITLE
reflection urls from project

### ISSUE THREAD

**Issue by dyanarose** [openapi-tools/swagger-maven-plugin#5]
Trying to use this with a project with swagger-core/swagger-jaxrs2 2.0.5 annotations.

I found the output to be missing every single api and only including the Info data from the plugin specification.

I've been debugging my way through this and I've found that when the Configuration Builder builds it doesn't include any of the URLs for the project, so it wouldn't be able to find any project classes under the listed resourcePackages.

I altered the plugin code to ensure that the plugin picks up the project URLs, and also had to set the config ClassLoaders.

This has then led to the project classes being loaded, but unfortunately I'm now at a dead end when a project class that very clearly is annotated with javax.ws.rs.Path (and it shows in the annotationData on reflection), responds to ReflectionUtils.getAnnotation(cls, javax.ws.rs.Path.class) with null.

Now, I assume I'm doing something quite wrong here (and I'm by no means a reflection guru) as my expectation is that the plugin would run during the build process, run over the classes found under the resourcePackages and an OAS 3.0 would come out the other side, similar to how the kongchen plugin works for OAS 2.0, but it doesn't seem to want to play ball for me.

Plugin configuration in case you're curious
```
         <plugin>
                <groupId>io.openapitools.swagger</groupId>
                <artifactId>swagger-maven-plugin</artifactId>
                <version>2.0.0</version>
                <configuration>
                    <resourcePackages>
                        <resourcePackage>com.my.api</resourcePackage>
                    </resourcePackages>
                    <outputFormats>
                        <outputFormat>JSON</outputFormat>
                        <outputFormat>YAML</outputFormat>
                    </outputFormats>
                    <outputDirectory>target/api</outputDirectory>
                    <outputFilename>swagger</outputFilename>
                </configuration>
            </plugin>
```

(edits for language, too much reflection debugging seems to remove my ability to type)

**Comment by langecode** [comment]
Thank you for reporting this. Very nice with such a detailed description. I will take a look ASAP. The plugin actually has some integration tests which I had expected would also catch such classloading issues but apparently not. I am maintaining a similar Gradle plugin and there I have Seem a similar issue relating to transitive dependencies of Swagger core jax-rs module having changed.

**Comment by langecode** [comment]
I have tried to reproduce this in a simple JAX-RS based project but I cannot seem to get this behaviour. My suspicion would be that this is something related to the Maven dependencies providing the JAX-RS annotations. Would it be possible for you to post the entire `pom.xml`? I expect your project imports either `java-ee-web-api`, `javax.ws.rs-api` or similar - is that set to `provided` scope in your project?

**Comment by dyanarose** [comment]
I'll get the pom to you by the end of the week (holiday 🙂 )

**Comment by dyanarose** [comment]
there are no dependencies marked as provided, no.

Pom can be seen at: https://gist.github.com/dyanarose/0925e6ae01e82e59f80a227debaef36c

**Comment by langecode** [comment]
Hard one to reproduce this one. Tried setting up at test project with the same dependencies but I still get an OpenAPI (Swagger) document with all the correct endpoints. I am wondering whether running Maven with "-X" would get any useful output but otherwise I am afraid we will have to see if we can create a simple project (without your most properly confidential classes :-) ) to reproduce the issue.

I cna understand why your dependencies are not scoped "provided" as you do not deploy in an application container but generate an executable jar.

**Comment by dyanarose** [comment]
Hmmm, indeed. I'll get together a small sample project and stick it in a Dockerfile.

Could be environmental as well.

**Comment by langecode** [comment]
Well, I have managed to reproduce the error now. Seems there is a difference in how the classpath for the plugin depending on whether packaging is `war` or `jar` - until now we have always been building a `war` file for deployment into an application container. So in essence the classes of the Maven module itself is not exposed to the plugin.

I will look into the classpath setup of the plugin however I do have a workaround for you. If you add the project itself as dependency of the plugin it seems to generate the OpenAPI document.

```xml
            <plugin>
                <groupId>io.openapitools.swagger</groupId>
                <artifactId>swagger-maven-plugin</artifactId>
                <version>2.0.0</version>
                <dependencies>
                    <dependency>
                        <groupId>com.salecycle</groupId>
                        <artifactId>dashboards-api</artifactId>
                        <version>${project.version}</version>
                    </dependency>
                </dependencies>
                <configuration>
                    <resourcePackages>
                        <resourcePackage>com.salecycle.dashboardsapi.api</resourcePackage>
                    </resourcePackages>
                    <outputFormats>
                        <outputFormat>JSON</outputFormat>
                        <outputFormat>YAML</outputFormat>
                    </outputFormats>
                    <outputDirectory>target/api</outputDirectory>
                    <outputFilename>swagger</outputFilename>
                </configuration>
                <executions>
                    <execution>
                        <phase>compile</phase>
                        <goals>
                            <goal>generate</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
```

**Comment by dyanarose** [comment]
👍 yup, confirmed that adding the dependency does the trick for now. Thanks for digging through with this.

**Comment by langecode** [comment]
This should be fixed in release [2.0.1](https://github.com/openapi-tools/swagger-maven-plugin/tree/v2.0.1).
